cli/flox/src/commands/install.rs

```diff
@@ -232,16 +270,38 @@ impl Install {
             message::warning(warning);
         }
 
-        // Print which new packages were installed
-        for pkg in packages_to_install.iter() {
-            if let Some(false) = installation.already_installed.get(pkg.id()) {
-                message::package_installed(pkg, &description);
-            } else {
-                message::warning(format!(
-                    "Package with id '{}' already installed to environment {description}",
-                    pkg.id()
-                ));
-            }
+        let installed = if let Some(packages_retried) = packages_retried {
+            packages_retried
+        } else {
+            packages_to_install
+                .iter()
+                .map(|p| PackageToInstallRetry {
+                    partial: false,
+                    pkg: p.clone(),
+                })
+                .collect::<Vec<_>>()
+        };
+        let partitioned =
+            Self::partition_installed_packages(&installed, &installation.already_installed);
+
+        // Print status messages for the installation attempt
+        if !partitioned.successes.is_empty() {
+            message::packages_successfully_installed(&partitioned.successes, &description);
+        }
+        for pkg in partitioned.partials.iter() {
+            message::warning(format!(
+                "'{}' installed only for the following systems: {}",
```

Comment:
I'm surprised that we didn't have any existing test coverage for this messaging change.

Can you add some tests, probably using the new tracing filter rather than Bats?

Reply:
I was also surprised, but I assumed that it just fell into the category of "we decided it wasn't worth it to test some messages." Anecdotally, nothing failed when I accidentally printed the systems without unwrapping and generated a `Some([....])`.